nopassword/models.py

```diff
@@ -64,7 +64,9 @@ def create_code_for_user(cls, user, next=None):
         if not user.is_active:
             return None
 
-        code = cls.generate_code(length=getattr(settings, 'NOPASSWORD_CODE_LENGTH', 20))
+        code = None
+        while not cls.code_is_valid(code):
```

Comment:
I am a bit unsure about the while loop though, however unlikely that it will be an existing code several times there maybe should be an escape hatch?